modules/survey_module/jsx/DirectEntry.js

```diff
@@ -0,0 +1,384 @@
+				/>
+			);
+		}
+			
+		if (this.state.page === this.state.InstrumentJSON.Elements.length) {
+			buttons = (
+				<div>
+				 	<button type="button" className="btn btn-primary btn-lg" onClick={this.prevPage}>Prev</button>
+				 	<button type="button" className="btn btn-primary btn-lg" onClick={this.submit}>Submit</button>
+				 </div>
+			 );
+		} else if (this.state.page === -1 || (this.state.page === 0 && this.state.InstrumentJSON.Elements.length === 1)) {
+			 buttons = (
+			 	<button type="button" className="btn btn-primary btn-lg">Done</button>
+			 );
+		} else if (this.state.page === 0) {
+			buttons = (
+			 	<button type="button" className="btn btn-primary btn-lg" onClick={this.nextPage}>Next</button>
+			 );
+		} else if (this.state.page === this.state.InstrumentJSON.Elements.length - 1) {
+			buttons = (
+				<div>
+				 	<button type="button" className="btn btn-primary btn-lg" onClick={this.prevPage}>Prev</button>
+				 	<button type="button" className="btn btn-primary btn-lg" onClick={this.nextPage}>Done</button>
+				 </div>
+			 );
+		} else {
+			buttons = (
+				<div>
+				 	<button type="button" className="btn btn-primary btn-lg" onClick={this.prevPage}>Prev</button>
+				 	<button type="button" className="btn btn-primary btn-lg" onClick={this.nextPage}>Next</button>
+				</div>
+			 );
+		}
+		const style = {
+			width: this.state.completionStats.completed / this.state.completionStats.total * 100 + '%'
+		}
+		return (
+			<div>
+				<nav className="navbar navbar-default navbar-fixed-top">
+  					<span className="h1" className="navbar-brand">LORIS</span>
+				</nav>
+				<div id='page' className='container-fluid'>
+					{DirectEntryFormElements}
+					<div className='question-container col-xs-12 col-sm-10 col-sm-offset-1'>
+						{buttons}
+					</div>
+				</div>
+				<div className="navbar navbar-default navbar-fixed-bottom">
+					<div className="col-xs-5 footer-bar">
+						{this.state.completionStats.completed} of {this.state.completionStats.total} Answered
+					</div>
+				    <div className="col-xs-4 footer-bar">
+				      	<div className="progress">
+						  <div className="progress-bar-info progress-bar-striped" role="progressbar" aria-valuenow="60" aria-valuemin="0" aria-valuemax="100" style={style}>
+						    &nbsp;
+						  </div>
+						</div>
+				    </div>
+			    </div>
+			</div>
+		);
+	}
+}
+
+class ReviewPage extends React.Component {
+	constructor(props) {
+		super(props);
+	}
+
+	render() {
+
+		let questions = this.props.reviewData.questions.map((element) => {
+			console.log(element);
+			return (
+				<tr className='reviewPage'>
+					<td>{element.question}</td>
+					<td>{element.response}</td>
+				</tr>
+			);
+		});
+
+		return (
+			<div className='question-container col-xs-12 col-sm-10 col-sm-offset-1'>
+				<h3>Review You Submission</h3>
+				<table className="table table-striped table-bordered">
+					
+					<tbody>
+					{questions}
+					</tbody>
+				</table>
+			</div>
+		)
+	}	
+}
```

Comment:
please add proptypes so that reviewData is described properly.

Reply:
Default props would also be useful.